File: remove_duplicates_from_sorted_array#26/solution_1.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  len = 0
  nums.each_with_index do |num, index|
    if index == 0 or nums[len-1] != num
      nums[len] = num
      len += 1
    end
  end
  len
end

File: remove_duplicates_from_sorted_array#26/solution_1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  let len = 0;
  for(let i = 0; i < nums.length; i++) {
    if (!i || nums[len-1] != nums[i]) {
      nums[len++] = nums[i];
    }
  }
  return len;
};
